### PR TITLE
Address empty patches

### PR DESCRIPTION
**Pull request by masterleinad**
We need to make sure that we can deal with the case that MPI processes don't own any patch.

### CODE REVIEW

**Comment by Rombur** [comment]
That just looks wrong to me. How is it possible that a process doesn't have any agglomerate? The only case where it is not a bug it's if the process doesn't own any cells.

**Comment by masterleinad** [comment]
> The only case where it is not a bug it's if the process doesn't own any cells.

Yes, exactly. For me that happens with one global refinement and two MPI processes.

File: source/dealii/dealii_hierarchy_helpers.cc [hunk 1]
```diff
@@ -112,7 +112,9 @@ DealIIHierarchyHelpers<dim, VectorType>::build_restrictor(
         delta_correction_acc;
     bool is_halo_agglomerate = false;
     unsigned int const n_local_eigenvectors =
-        delta_correction_matrix.m() / interior_agglomerates.size();
+        interior_agglomerates.empty()
+            ? 0
+            : delta_correction_matrix.m() / interior_agglomerates.size();
```
Comment (Rombur) [inline source/dealii/dealii_hierarchy_helpers.cc]: Shouldn't we just return at this point

File: source/dealii/dealii_matrix_free_hierarchy_helpers.cc [hunk 2]
```diff
@@ -109,7 +109,9 @@ DealIIMatrixFreeHierarchyHelpers<dim, VectorType>::build_restrictor(
         delta_correction_acc;
     bool is_halo_agglomerate = false;
     unsigned int const n_local_eigenvectors =
-        delta_correction_matrix.m() / interior_agglomerates.size();
+        interior_agglomerates.empty()
+            ? 0
+            : delta_correction_matrix.m() / interior_agglomerates.size();
```
Comment (Rombur) [inline source/dealii/dealii_matrix_free_hierarchy_helpers.cc]: Same here

**Comment by aprokop** [comment]
Could a better approach be to do a comm split on the finest level, and just work with MPI processes that have at least one cell?

**Comment by masterleinad** [comment]
I am not sure how much work that is.

**Comment by Rombur** [comment]
Unless you really want to change the communicator @aprokop, I think we can merge the current PR

**Pull request merged**